resources/js/charts/changelog-chart.ts

```diff
@@ -0,0 +1,308 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the GNU Affero General Public License v3.0.
+// See the LICENCE file in the repository root for full licence text.
+
+import * as d3 from 'd3';
+import { first, groupBy, kebabCase, last, map, sumBy } from 'lodash';
+import moment from 'moment';
+import { fadeIn, fadeOut } from 'utils/fade';
+import { formatNumber } from 'utils/html';
+import { parseJson } from 'utils/json';
+
+interface BuildHistory {
+  created_at: string;
+  label: string | null;
+  normalized: number;
+  user_count: number;
+}
+
+interface ChartConfig {
```

Comment:
what I meant was calling something that contains actual data "config" isn't quite right